pkg/argoapp/argoapp.go

```diff
@@ -117,3 +120,30 @@ func NewApplication(config ApplicationConfig) (*unstructured.Unstructured, error
 
 	return &unstructured.Unstructured{Object: obj}, nil
 }
+
+func NewApplication(config ApplicationConfig) (*v1alpha1.Application, error) {
+	obj, err := NewUnstructuredApplication(config)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	app, err := UnstructuredToArgoApplication(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	return app, nil
+}
+
+func UnstructuredToArgoApplication(u *unstructured.Unstructured) (*v1alpha1.Application, error) {
```

Comment:
I wonder if that really should be exported.

Reply:
TODO